pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -60,14 +60,7 @@ func newS3Agent(accessKey, secretKey, endpoint, region string, debug bool, tlsCe
 	tlsEnabled := false
 	if len(tlsCert) > 0 || insecure {
```

Comment:
If I pass `tlscert` but set `insecure` to `true`(bucket health checker for self-signed certificate), it results in `tlsEnabled` to `true` with insecure flag set. Do it fails for the session.NewSession() ? I mean due to conflicting values in `http transport type` and  `WithDisableSSL` values.

Reply:
I think it's a good point and I feel like both are conflicting anyway. Should we stop using `WithDisableSSL` since we pass our own HTTP Client?

Reply:
FYI `WithDisableSSL` sets the HTTP scheme to either HTTP or HTTPS so I think providing a TLS config through the HTTP Client adds up...

Reply:
At the end, we can still use `WithDisableSSL` to use the right scheme but specify that the client should ignore the certificate hostname validation with `InsecureSkipVerify`.

Makes sense?